components/admin_console/group_settings/group_details/group_details.jsx

```diff
@@ -34,34 +34,39 @@ export default class GroupDetails extends React.PureComponent {
             link: PropTypes.func.isRequired,
             unlink: PropTypes.func.isRequired,
             patchGroupSyncable: PropTypes.func.isRequired,
+            patchGroup: PropTypes.func.isRequired,
         }).isRequired,
     };
 
     static defaultProps = {
         members: [],
         groupTeams: [],
         groupChannels: [],
-        group: {display_name: ''},
+        group: {display_name: '', allow_reference: false},
         memberCount: 0,
     };
 
     constructor(props) {
         super(props);
+        const group = props.group;
         this.state = {
             loadingTeamsAndChannels: true,
             addTeamOpen: false,
             addChannelOpen: false,
+            allowReference: Boolean(group.allow_reference),
         };
     }
 
     componentDidMount() {
         const {groupID, actions} = this.props;
```

Comment:
could change this to `const {groupID, actions, group} = this.props;` 
And then modify line 69 to be
```
this.setState({group, allowReference: Boolean(group.allow_reference)});
```